brion/compartmentReport.cpp

```
@@ -122,12 +124,27 @@ size_t CompartmentReport::getBufferSize() const
     return _impl->plugin->getBufferSize();
 }
 
-floatsPtr CompartmentReport::loadFrame(const float timestamp) const
+std::future<floatsPtr> CompartmentReport::loadFrame(const float timestamp) const
+
 {
-    if (timestamp < getStartTime() || timestamp > getEndTime())
-        return floatsPtr();
+    auto task = [timestamp, this] {
+        if (timestamp < getStartTime() || timestamp > getEndTime())
+            return floatsPtr();
```

As said in the app, throw instead of returning an empty array?